packages/dashboard/src/components/actions/settings.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+
+import { Box, Checkbox, Modal, NonCancelableCustomEvent, SpaceBetween } from '@cloudscape-design/components';
+import { BaseChangeDetail } from '@cloudscape-design/components/input/interfaces';
+
+import LabeledInput from '../util/labeledInput';
+import { useGridSettings } from './useGridSettings';
+
+// Should never return NaN
+const numberFromDetail = (event: NonCancelableCustomEvent<BaseChangeDetail>) => parseInt(event.detail.value) || 0;
```

Comment:
Nit: I believe linters are starting to mark this is an error and recommending `??` used instead of short-circuiting.

Reply:
I don't know if that would work in this scenario since NaN is not null / undefined? But I know it's falsey :D